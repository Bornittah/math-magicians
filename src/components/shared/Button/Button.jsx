import React from 'react';
import PropTypes from 'prop-types';

class Button extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { className, text } = this.props;
    return (
      <button type="submit" className={className}>
        {text}
      </button>
    );
  }
}

Button.defaultProps = {
  className: '',
  text: '',
};
Button.propTypes = {
  className: PropTypes.string,
  text: PropTypes.string,
};
export default Button;
