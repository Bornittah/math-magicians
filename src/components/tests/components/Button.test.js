import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Button from '../../shared/Button';

describe('Testing Button component', () => {
  const onClick = jest.fn();
  it('Testing with snapshot', () => {
    const tree = renderer.create(<Button
      className="button gray-bg flex-1"
      value="7"
      onClick={onClick}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Testing clickEvent', () => {
    render(<Button
      className="button gray-bg flex-1"
      value="7"
      onClick={onClick}
    />);
    fireEvent.click(screen.getByText('7'));
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
