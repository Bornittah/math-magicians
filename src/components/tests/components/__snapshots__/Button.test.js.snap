// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component Testing button component 1`] = `
<button
  className="button gray-bg flex-1"
  onClick={[MockFunction]}
  type="button"
  value="7"
>
  7
</button>
`;

exports[`Testing Button component Testing with snapshot 1`] = `
<button
  className="button gray-bg flex-1"
  onClick={[MockFunction]}
  type="button"
  value="7"
>
  7
</button>
`;
